### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by KristinaGru**

### CODE REVIEW

File: javascript/javascript1/week2/full_name.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Flight booking fullname function
+
+function getFullName(firstName, surname, gender) {
+  if (
+    firstName === '' ||
+    typeof firstName != 'string' ||
+    surname === '' ||
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/full_name.js:7]: what if we pass just spaces? (like "      ")

File: javascript/javascript1/week2/candy_helper.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  let sum = 0;
+  while (i < prices.length) {
+    sum += prices[i];
+    i++;
+  }
+  if (sum < amountToSpend) {
+    return 'You can buy more, so please do!';
+  } else {
+    return 'Enough candy for you!';
+  }
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candy_helper.js:37]: You can use ternary here

File: javascript/javascript1/week2/candy_helper.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      boughtCandyPrices.push(candyWeight * 0.03);
+      break;
+    default:
+      console.log('No such candy type');
+      break;
+  }
+}
+
+let amountToSpend = Math.random() * 100;
+amountToSpend = Math.round(amountToSpend * 100) / 100;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candy_helper.js:24]: You do not need to reassign this variable, you can do it in one line

File: javascript/javascript1/week2/candy_helper.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    return 'You can buy more, so please do!';
+  } else {
+    return 'Enough candy for you!';
+  }
+}
+
+window.onload = () => {
+  const paragraph = document.getElementById('spend');
+  paragraph.textContent = ` You can spend ${amountToSpend} DKK`;
+};
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candy_helper.js:43]: Nice using window.onload, but I think that you might want to use it wrapping all the DOM manipulations

**Review by YurijPaharev — CHANGES_REQUESTED** [review]
It is a magnificent and brilliant code! Still there is some place for improvement, please go through my comments

**Comment by KristinaGru** [comment]
> It is a magnificent and brilliant code! Still there is some place for improvement, please go through my comments

Thanks so much for all the feedback! I'll get right to it 🙌

**Review by YurijPaharev — APPROVED** [review]
It is great how far you are proceeding, I wouldn't do it better myself.
One thing though in the getFullName function you are using a long check `firstName === '' || typeof firstName !== 'string' || !firstName.trim()`. You can shorten it by just writing `typeof firstName !== 'string' || !firstName.trim()`. Because in this way you are checking that this value is indeed a string and after you are checking whether it has value or not, because string with spaces is converted to true, after we use trim() - it is converted to just empty string and `!''` is also converted to true. I encourage you to try it out and see results by yourself

**Comment by KristinaGru** [comment]
> It is great how far you are proceeding, I wouldn't do it better myself. One thing though in the getFullName function you are using a long check `firstName === '' || typeof firstName !== 'string' || !firstName.trim()`. You can shorten it by just writing `typeof firstName !== 'string' || !firstName.trim()`. Because in this way you are checking that this value is indeed a string and after you are checking whether it has value or not, because string with spaces is converted to true, after we use trim() - it is converted to just empty string and `!''` is also converted to true. I encourage you to try it out and see results by yourself

I can definitely see the logic there! Thanks for all the notes and the encouragement, I really appreciate it

**Pull request merged**